Confidence/src/main/java/com/spotify/confidence/Confidence.kt

```diff
@@ -60,11 +63,33 @@ class Confidence internal constructor(
         }
     }
 
-    suspend fun awaitReconciliation() {
-        if (currentFetchJob != null) {
+    suspend fun awaitReconciliation(timeoutMillis: Long = 5000) {
+        val delayTime = 3L
+        if (timeoutMillis <= delayTime) error("timeoutMillis need to be larger than $delayTime")
+        debugLogger?.logMessage("reconciliation started")
+        withSafeTimeout(delayTime) {
+            while (currentFetchJob == null) {
+                delay(1)
+                debugLogger?.logMessage("delayed")
```

Comment:
Why do we need to delay if there's currently no fetch jobs scheduled?

Reply:
Good comment and this made us take a closer look together and come up with a better solution.